NativeAOT-LLVM: bring IL backend inline with scanner - use unconstructed type for unbox

yowl opened the pull request:
This PR brings the EEType used for `unbox` in line with the scanner at 

https://github.com/dotnet/runtimelab/blob/20f3949f190f196ab4b958cacbeca1ff76ba3383/src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs#L918

We now longer use the `MaximallyConstructableType` type but the `NecessaryTypeSymbol`

Fixes the error at step 8 of https://github.com/dotnet/runtimelab/issues/1992 although attempting to run the produced wasm still fails with a different problem which looks related to the construction of the generic dictionaries

```
E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:148
      throw ex;
      ^

RuntimeError: memory access out of bounds
    at __GenericLookupFromType_S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>_TypeHandle_TKey_System___Canon (<anonymous>:wasm-function[605]:0x9cd6e)
    at S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor_2 (<anonymous>:wasm-function[602]:0x9c7ed)
    at S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor (<anonymous>:wasm-function[3520]:0x3cdc43)
    at S_P_CoreLib_System_AppContext__SetData (<anonymous>:wasm-function[296]:0x3dd2e)
    at Internal_CompilerGenerated__Module___SetAppContextSwitches (<anonymous>:wasm-function[3103]:0x3b74e6)
    at StartupCodeMain (<anonymous>:wasm-function[182]:0x16ecc)
    at __managed__Main (<anonymous>:wasm-function[3032]:0x3b2ba5)
    at main (<anonymous>:wasm-function[9304]:0x62c6b5)
    at E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:983:22
    at callMain (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7476:15)
    at doRun (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7530:23)
    at run (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7545:5)
    at runCaller (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:7453:19)
    at removeRunDependency (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:904:7)
    at receiveInstance (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:1071:5)
    at receiveInstantiationResult (E:\tmp\NativeAOTLLVMTest\bin\Debug\net7.0\browser-wasm\native\NativeAOTLLVMTest.js:1089:5)
```

kant2002 committed:
Another bunch of nullability fixes (#1572)

* Fix CS8601

* More CS8600

* Fix CS8605

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

MichalStrehovsky committed:
Log progress when writing object file with `--verbose` (#1576)

dotnet-bot committed:
Localized file check-in by OneLocBuild Task (#59286)

verdie-g committed:
Fix indentation in api prososal template (#59318)

5 spaces were used when 4 is the convention.

MichalStrehovsky committed:
Remove FixedArgument from metadata schema (#1577)

SchemaDef.cs is the meat of this change, most of the rest is just generated code and a small adjustment on the reader and writer side.

* `FixedArgumentAttributes` (`FixedArgument.Flags`) was unused
* The type of a fixed argument is redundant with the constructor signature (and is therefore not captured in the ECMA format). We shouldn't duplicate it.
* `NamedArgument` having a `FixedArgument` field was an odd shortcut.

MichalStrehovsky committed:
Do not generate unused metadata (#1578)

There are no runtime APIs to query things like custom attributes on a MemberRef. Remove them from schema. We were paying extra cost in each record to say "there are no custom attributes".

Also removing MethodImpl metadata. This is used by the runtime type system when we support metadata loading. Not something we do in NativeAOT.

Together with https://github.com/dotnet/runtimelab/pull/1577 shrinks the metadata blob by 6%.

MichalStrehovsky committed:
Document the reflectedonly mode (#1579)

kronic committed:
Bring XmlReader{Async} to one behavior (#51736)

* Replace XmlUntypedStringConverter to XmlUntypedConverter

* Add more test

* add more tests

* remove unssoprt test

* feedback review

Co-authored-by: kronic <kronic@softland.ru>

pavelsavara committed:
cmake list separator (#59317)

premun committed:
Use PowerShell in Android arm64 XHarness Helix payloads (#59340)

dotnet-maestro[bot] committed:
[main] Update dependencies from 4 repositories (#58999)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Premek Vysoky <premek.vysoky@microsoft.com>

7rakir committed:
Use correct NumberStyle when parsing double and float (#59205)

* User correct NumberStyle when parsing double and float

IParseable<float> and IParseable<double> would fail when parsing
a decimal point number due to incorrect NumberStyles parameter.

E.g. IParseable<double>.TryParse would fail as opposed to
IParseable<double>.Parse because the former explicitly states to use
NumberStyles.Integer

* Adding tests covering parse for float/double/half

* Delete MSBuild_pid-37116_6904092b1c9a481da960c9e28eba2a27.failure.txt

Co-authored-by: Tanner Gooding <tagoo@outlook.com>

MichalStrehovsky committed:
Ensure metadata addressing limit at compile time (#1580)

BrzVlad committed:
[mono][interp] Add checks for endfinally opcode (#59332)

* [interp] Add checks for endfinally opcode

Throw invalid IL if endfinally is not within a finally block.

* [tests] Re-enable test

SingleAccretion committed:
Refactor subrange assertion generation and propagation (#55186)

* Refactor subrange generation

Move it into its own function.

No diffs.

* Stop generating assertions for TYP_U/INT ranges

They are not useful: they do not tell us anything about the value.
They can be harmful: code operating on the assertions can wrongly
remove a checked cast because of an assertion for a non-checked one.

* Add the header for optTryExtractSubrangeAssertion

* Refactor subrange assertions

This change completely refactors the way subrange assertions are
generated and consumed. It rationalizes and generalizes the model
for subrange assertions which are generated and propagated for casts,
makes AssertionDsc::Range into an actual abstraction, eliminates bugs
related to wrong assertions generated for TYP_BOOL (without CQ loss),
enables propagation for checked casts to non-small types both on 64
and 32 bit, all the while making the AssertionDsc structure smaller.

There are quite a few moving parts, but at the heart of this change
are following items:

1. Make ranges symbolic, i. e. made up of symbolic upper and lower
   bounds, represented by the new SymbolicIntegerValue enum class.
   This is necessary to not regress memory consumption and throughput
   of operations on 32 bit while handling long subranges.
2. Split the concept of "input" and "output" for a cast node.
   To properly handle checked cast and make the code "obviously correct",
   it is necessary to split the two values, one the cast node produces
   and one it consumes, apart. The old code only handles the "easy" case
   of the "input" and "output" ranges matching (such is the case for small
   types), the new code must be, and is, robust in reasoning about, e. g.
   CAST_OVF(uint <- ulong) vs CAST_OVF(int <- long). The ranges that
   represent these values are consistently in the "signed" format, making
   handling them easy.
3. Tweak the propagation code to not remove representation-changing casts,
   make it explicit and obvious in what cases (and why) it retypes the
   local under the cast.

As a side-effect of the above changes, TYP_BOOL is now treated as TYP_UBYTE
in propagation, as it should be. Actually, the old code didn't remove
redundant casts to TYP_BOOL, the new code does, which is also where the
bulk of the positive diffs come from.

The old code also generated "useless" ranges like [INT_MIN..INT_MAX], the new
code only generates ranges that could be used in subsequent propagation.

Also, some miscellaneous cleanup, such as adding function headers, was performed.

* Use correct ranges for global subrange assertions

In global assertion propagation, tentative assertions are
created for casts, with the expectation that proving them
via implication will allow for removing the cast.

The old code did this through the common path with local
assertions, however, this cannot be done the same way anymore
because the two types of assertions are fundamnetally different.

In local assertion propagation, subranges are derived from ASGs,
where the cast is on the RHS and the local - LHS, so the "output"
range for the cast must be used - local is assigned value the cast
**produces**.

In contrast, global subrange assertions say that the value the
cast **consumes** is within the correct bounds for the cast to
become a no-op or non-overflowing.

Because of this fundamnetal distinction, I chose to make global
assertions for casts in a new method instead of plumbing it
through "optCreateAssertion". Note that it would be possible to
do it via "optCreateAssertion" as well - global propagation
always passes OAK_SUBRANGE to it, while local - OAK_EQUAL, but
I decided against that option for the reason that it seems
unintuitive to me that it would work the way it would. These
tentative assertions are a very special case, so I think it is
better to treat them that way. This leads to some minor code
duplication, but I think that's ok.

No diffs for this commit, though that's expected as "proving"
the generated assertion is quite rare.

* Refactor AssertionDsc::Range into its own class

Make IntegralRange the owner of the logic for various
methods relating to it and decouple it from AssertionDsc.

IntegralRange will be useful in the future for other code
that deals with cast removal, e. g. in fgMorphCast.

Also some miscellaneous formatting fixups.

* Change the implementation of SymbolicToRealValue

Using a static array results in smaller and more
efficient code on all compilers.

* FP & GC castees in IntegralRange::ForCastOutput

This change is for completeness, since "fgMorphCast" decomposes
casts from FP types to small types today and as such assertions
are generated for them anyway.

No diffs for this commit as expected.

* Add the TYP_BOOL quirk

To avoid regressions.

* Fix the input ranges for non-overflowing casts

They did not handle "long" cases and as such were too conservative.

* Add missing function headers

* Remove a TODO now that BOX import has been fixed

* Use gtEffectiveVal & ChangeType to simplify code

* Fix build break

* Add a more detailed comment

* Tweak the comment

SingleAccretion committed:
Delete dead code (#56865)

* Delete dead QMARK optimization

It is checking for both sides of GT_COLON being identical,
but QMARKs are only created in some very limited circumstances
today, and certainly not with identical then and else clauses.

No diffs.

* Delete dead code from fgSetTreeSeqHelper

QMARKs are rationalized before it is ever called.
Likewise, GT_INDs with "a special structure" are a thing of the past.

No diffs.

* Delete unused variable

* Delete unused impAssignSmallStructTypeToVar

* Delete unused gtIsVtableRef & getObjectHandleNodeFromAllocation

* Delete unused IsTreeAlwaysHoistable

* IsAddWithI32Const

* Delete unused Jit config values

* Delete unused fgEstimateCallStackSize

* Delete unused parameters in fgMorphStmts

* Delete unused StripExOp

elinor-fung committed:
Remove some unnecessary allocations / copies on startup (#59270)

am11 committed:
Ignore MSBuild_Logs/ directory (#59323)

This is a new directory used by msbuild for logging since https://github.com/dotnet/msbuild/commit/cdb5077c451180ab38161e0b5e70f5448e70355b.

Commit:
Suppress SendPacketsElement_FileStreamMultiPartMixed_MultipleFileStreams_Success test on windows 11 (#59203)

DrewScoggins committed:
Change logic to not break private runs (#59356)

The previous change here broke private runs as it required the channel map
to have an entry matching the name of the runtime branch you were testing.
This fix now defaults the channel to main for private runs.

AlexRadch committed:
Microsoft.VisualBasic.Core.Tests fail tests with NaN conversion (#59321)

* Microsoft.VisualBasic.Core.Tests fails tests with NaN conversion

* Update src/libraries/Microsoft.VisualBasic.Core/tests/SingleTypeTests.cs

Co-authored-by: Dan Moseley <danmose@microsoft.com>

elinor-fung committed:
Delete duplicate runtime tests (#58676)

MichalStrehovsky committed:
Fix typo (#1583)

thaystg committed:
[wasm][debugger] Reverting the old behavior of scope id numeration (#59368)

* Keeping the old behavior of scope id what we have before start using debugger-agent.

* Fix compilation.

* Fix compilation.

vargaz committed:
[mono][wasm] Rework the handling of GC references in AOTed code. (#59352)

Previously, variables holding GC refs were marked volatile so they
were loaded/stored to the C stack on every access. Since the stack
is conservatively scanned, all the objects pointed to by it are pinned,
so there is no need to load them on every access. Instead of
marking them as volatile, allocate a 'gc pinning' area on the stack,
and store ref variables to it after they are assigned. This improves
code size and performance.

BrzVlad committed:
[mono][interp] Don't assert when encountering invalid IL opcode (#59334)

* [mono][interp] Don't assert when encountering invalid IL opcode

Throw exception instead.

* [tests] Re-enable tests

MichalStrehovsky committed:
Fix building GC on Android (#58703)

ManickaP committed:
MmQuicListener Comment (#59365)

Maximys committed:
Fix invalid parsing of not unicode Uri (#59252)

* Fix parsing of not unicode Uri (#56482).

* Changing test for reproducing problem with Uri creation by Unicode (#59252).

jakobbotsch committed:
Allow contained indirections in tailcalls on x64 (#58686)

This adds support for contained indirections in tailcalls on x64. The
significant diff of this change is refactoring to be able to reuse the
code for generating call instructions when generating tailcalls as well.
Other than that, the main change is to allow contained indirs in
lowering and to ensure LSRA uses volatile registers for the addressing
mode so that the registers are not overridden by the epilog sequence.

To be sure we insert rex. prefix correctly I also refactored the emitter
to base the decision on a new instruction (INS_i_jmp) and emit it when
necessary. The rex. prefix is necessary because the unwinder uses it to
determine that a tail jmp is non-local and thus part of the epilog.

Finally, unlike the OS unwinder our unwinder needs to compute the size
of the epilog. This computation was wrong for jmp [foo] for addressing
modes/rip-relative addressing, so fix this as well. Presumably that has
not been a problem before since we did not generate these instructions
in managed code (although native code may have had these instructions --
not sure if we use that unwinder for native code).

josteink committed:
Fix crossgen for FreeBSD by adding FreeBSD RID. (#59331)

* Fix crossgen for FreeBSD by adding FreeBSD RID.

Fix implemented as described by @Thefrank in this thread:
https://github.com/dotnet/runtime/issues/14537#issuecomment-922546527

* Add FreeBSD platform-support to crossgen2.

AaronRobinsonMSFT committed:
Check if External Object Context is still active after a possible GC (#59285)

* Handle a possible GC after allocating an array by iterating over the hash again.

Co-authored-by: Elinor Fung <elfung@microsoft.com>

Commit:
<EnablePreviewFeatures> for HttpStress (#59410)

viewizard committed:
[x86/Linux] Fix SIGSEGV on Debugger.Break() during debugging. (#59350)

SIGSEGV occur in src/vm/i386/cgenx86.cpp `HelperMethodFrame::UpdateRegDisplay()` at line `pRD->pCurrentContext->Eip = pRD->ControlPC = m_MachState.GetRetAddr();`, since `pRD->pCurrentContext` is NULL.
The point of issue - wrong code block compilation in DebuggerWalkStack() in case of Linux x86.

github-actions[bot] committed:
[main] [Release/6.0]fix the package version for non-shipping csproj packages (#59388)

* fix the package version for non-shipping package

* use version prefix

Co-authored-by: Anirudh Agnihotry <anagniho@microsoft.com>

davidwrighton committed:
Fix gtCloneExpr when cloning during R2R compilation a GT_ALLOCOBJ node (#59395)

* Fix gtCloneExpr when cloning during R2R compilation a GT_ALLOCOBJ node
- Without this fix cloned expressions with allocations will fail
- This is most common in profile guided code around devirtualization, but I believe it can occur in other where gtCloneExpr is used
- Symptom of the failure is a compilation failure during crossgen2

cshung committed:
Update Unloadability documentation (#59383)

t-mustafin committed:
[unix x86] Fix GCMemoryInfoData size assertion (#59417)

davidwrighton committed:
Fix issues where the module index GUID table isn't properly computed (#59375)

* Fix issues where the module index GUID table isn't properly computed
- Since there are two possible places where the module indices must be up to date, ensure that they are finalized before either Node writes its data
- Ensure that module indices are precomputed when needed from inlinee data and the InstanceEntryPointTableNode

CarnaViire committed:
[HTTP/3] Fix NullReferenceException on Timeout when streaming request body (#59204)

Brings the logic for canceling streaming request body from HTTP/2 to HTTP/3.

The problem was that cancellation token was not applied to SendContentAsync, which resulted in Http3WriteStream not being disposed properly.

Fixes #57619

kant2002 committed:
First round of converting System.Data.OleDb to COMWrappers (#54822)

* First round of converting System.Data.OleDb to COMWrappers
This change just remove warning from ILLink and do not make attempt
to convert other COM object creation to use ComWrappers

Also I have to add target net5.0-windows where ComWrappers would be
used. Same concerns regarding testing as in
https://github.com/dotnet/runtime/pull/54636

Two calls inside OleDbDataAdapter call GetErrorInfo but do not do anything with results. That's probably mistake from previous refactoring. These parts does not touched and left as is. Need advice on that specific parts how to proceed.

* Update src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

* Apply PR recommendations

* Make closer to original

* One more change

* Apply PR feedback

* Update src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj

Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj

Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>

* Fix compilation errors

* Fix errors

* Consolidate code

* Fix compilation warnings

* Fix compilation error

* Fix compilation error

* Address PR feedback

* Update PR feedback

* Address PR feedback

* Direct cast

* Ask wrapper to implement IDisposable

* Update src/libraries/System.Data.OleDb/src/OleDbConnection.COMWrappers.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/DbPropSet.COMWrappers.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

* Restore changes in OnInfoMessage

* Consolidate different part of code

* Remove junk (probably from merge)

* Found that last commit was a mess

* Remove no longer needed annotations

* Apply suggestions from code review

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

* Update src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

* Free BSTR

* Remove reference to no longer used files in this PR

* Add data

* Fix compilation errors

* Last blow to compiler

Co-authored-by: Aaron Robinson <arobins@microsoft.com>
Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>
Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

thaystg committed:
Fixing memory corruption. (#59419)

kunalspathak committed:
Arm: Use correct form of store instruction when data is float (#59187)

* use the correct src type for str operation

* Add test case

* Proper fix

* Extend to arm64

* remove unnecessary property from csproj

* Update the test case

* restore the assert

* jit format

* Use ins_StoreFromSrc

* jit format

sandreenko committed:
Delete a legacy local for arm32. (#59380)

maryamariyan committed:
update xunit version (#59275)

uweigand committed:
Big-endian fixes: PDB creation by ilasm (#59422)

* Fix byte order of debug directory fields

* Fix byte order of PDB stream data (PdbHeap::SetData)

layomia committed:
Add notes about not calling src-gen APIs directly and make them non-browsable (#59385)

radical committed:
[wasm] Enable System.Text.Json.SourceGeneration.Tests (#59396)

PeterSolMS committed:
We should not call distribute_free_regions at the end of background GC - then it may run concurrently with user threads allocating and getting regions from the free lists. (#59413)

Commit:
CI matrix change: drop OSX 10.13 (#59084)

* Drop OSX 10.13 from CI matrix

* Drop OSX 10.14 from PR  CI matrix

Commit:
CI matrix change: ubuntu (#58987)

* Remove ubuntu 16.04

* Remove another occurence

PeterSolMS committed:
Fix brick table issue due to race condition in adjust_limit_clr (#59411)

With regions, we had observed AVs due to 0 brick table entries.

As it turns out, this is due to a race condition in adjust_limit_clr where ephemeral_region may change before we check it to determine whether we need to set bricks.

The fix is simply to fetch ephemeral_region into a local before we release the more space lock.

This PR also contains the code checking the brick table for 0 entries - I propose leaving this code enabled even in retail until we are confident the race condition mentioned above is the only cause.

thaystg committed:
[wasm][debugger] Fix concurrent access to cache. (#59412)

* Fix concurrent access to cache.

* Fixing test case.

* Changing what was suggested by @radical.

jkotas committed:
Cleanup reflection RuntimeType checks (#59457)

* Delete IsRuntimeImplemented helper method

No longer necessary

* Use compact syntax for RuntimeType checks

jkotas committed:
Replace IRuntimeImplemented with RuntimeType and RuntimeAssembly base types (#1586)

To match how reflection is structured in CoreCLR and Mono. It should be also a step towards eliminating System.Private.DisabledReflection.

ManickaP committed:
Retrieve dumps from stress pipeline (#58860)

* Removed parts of stress to make the test loop faster.

* Dump collection and artifacts publish

* Chown dumps

* Removed Windows build for tests

* Bringing windows on par with linux

* Fixed windows paths.

* reverted testing changes

* feedback

* Update eng/pipelines/libraries/stress/http.yml

Co-authored-by: Santiago Fernandez Madero <safern@microsoft.com>

* feedback renames

* feedback newlines

Co-authored-by: Santiago Fernandez Madero <safern@microsoft.com>

jkotas committed:
Delete Type.GetTypeHandleInternal (#59459)

Never-overriden virtual method forwarding to virtual property.

jkotas committed:
Merge remote-tracking branch 'upstream/runtime-main' into merge-main

jkotas committed:
Merge pull request #1587 from jkotas/merge-main

Merge from dotnet/runtime

jkotas committed:
Delete IsRuntimeImplemented (#1588)

jkotas committed:
Switch to shared System.Enum implementation (#1590)

MichalStrehovsky committed:
Align trimming settings with IL Linker (#1596)

Aligns the ILCompiler targets with ILLink.

It's all rather complex because:

* One can specify that an assembly is trimmable using a custom attribute. This doesn't mean it will be trimmed in IL Link, it just marks it.
* One can specify a default trimming action (keep everything or trim) on the command line for assemblies not marked as trimmable
* One can specify a trimming action for listed assemblies on the command line
* In ILLink, one can also specify trimming action for assemblies marked IsTrimmable. For ILCompiler, I'm making this assume the action is trim. ILLink can also say to root. I don't see much point.

I tested this with both .NET 5 and .NET 6 SDKs.

There's a difference in behavior - in the past, setting `TrimMode=link` would trim the whole app, but that's not what illinker does. For .NET 5, one has to do the [custom target dance](https://github.com/davidfowl/LinkedAspNetCoreApplication/blob/fd7454f23e7e48a4302fb239f72a914ce4c9dc73/LinkedAspNetCoreApplication/LinkedAspNetCoreApplication.csproj#L14-L21). On .NET 6, one can just specify TrimmerDefaultAction=link.

yowl committed:
bring IL backend inline with scanner

yowl commented:
Cc @SingleAccretion

SingleAccretion reviewed (approved):
Seems we have some infrastructural CI issues.

jkotas commented:
The error is "Image Build.Windows.10.Amd64.VS2019.Open doesn't exist in pool NetCore-Public"

Could you please update the CI configs to match dotnet/runtime main?

yowl committed:
update the 2 windows pools images to match upstream

yowl commented:
Looks like I need to upgrade VS from 2019->2022.

yowl committed:
update cmake -G to VS 2022 for LLVM build

yowl committed:
minor: update nativeaot doc for building LLVM

copy init-init-vs-env.cmd from runtime - bit of a guess this one

yowl committed:
Cherry pick 457796f3c76788c79bc2fa9b14cccd0d5b89f691 from runtime

and 4b6c7ed501b7e352d53898a2a0d2c2c5ddf3c34e

Get some VS 2022 changes, and the IlcSdkPath addition.

yowl committed:
back out defaultrooting which we dont have yet

yowl committed:
remove leading slashes for new git client

SingleAccretion commented on src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:
```diff
@@ -14,34 +14,33 @@ The .NET Foundation licenses this file to you under the MIT license.
 -->
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <PropertyGroup Condition="'$(TargetArchitecture)' == 'wasm'">
-    <NativeCodeGen>llvm</NativeCodeGen>
-  </PropertyGroup>
```
Doesn't this (and below) completely remove the build support for LLVM/WASM?

yowl replied:
oh yeah, thanks, will revert that

SingleAccretion commented:
Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.

yowl committed:
Use same fix as was done for MacOS

yowl committed:
reinstate LLVM Native targets

yowl commented:
> Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.



> Looks like we need #1596 with default rooting or some other way to prevent the tests from being trimmed.

Right, will merge up to that.  Looks like its just a week from where we are.

yowl committed:
Merge commit 'c2e2b8ff4a1841b8a48efec5d4137957a1144284' into unbox-type

yowl committed:
messed something up with the NativeObject extension as getting .obj when I should have .o.  Need to look more closely.

yowl committed:
Update Microsoft.NETCore.Native.targets

SingleAccretion reviewed (approved):
Latetst changes LGTM.
